### PR TITLE
Add support for building armv7 and aarch64 viewer/lsp binaries

### LINKED ISSUE #5707: No Linux ARM support for slint-lsp
As seen in the [Mason Registry](https://github.com/mason-org/mason-registry/blob/2024-07-27-dizzy-user/packages/slint-lsp/package.yaml), there is no ARM support for linux.
The LSP server repo itself specifies in the [readme](https://github.com/slint-ui/slint/tree/master/tools/lsp) that you need to use x86 binaries.
Yet, it looks like ARM binaries are available for macos.

Is this an oversight (granted, my setup of a linux inside chromeos is weird) or is there something blocking the build of such binaries ?

Thx

### PR DESCRIPTION
**Pull request by tronical**
Fixes #5707

### CODE REVIEW

**Comment by tronical** [comment]
Successful CI builds of lsp and viewer: https://github.com/slint-ui/slint/actions/runs/10142825544 https://github.com/slint-ui/slint/actions/runs/10142827874

**Review by ogoffart — COMMENTED** [review]
Only the name.is a bit strange. I wonder if the x86 one shouldn't also get a new name

**Review by ogoffart — APPROVED** [review]

**Comment by tronical** [comment]
> Only the name.is a bit strange.

I'll remove the "double" linux.

> I wonder if the x86 one shouldn't also get a new name

I was contemplating the same, but I don't want to change all the existing configs in editors, etc. that download this from our GH release...

File: .github/workflows/nightly_snapshot.yaml [hunk 1]
```diff
@@ -480,6 +480,12 @@ jobs:
             - uses: actions/download-artifact@v4
               with:
                   name: slint-lsp-linux
+            - uses: actions/download-artifact@v4
+              with:
+                  name: slint-lsp-linux-armv7-unknown-linux-gnueabihf
```
[the remark this answers is not in the document]
Comment (tronical) [reply .github/workflows/nightly_snapshot.yaml:485]: It turn out, the artifact has the correct name, this here is the wrong name to download :)

**Pull request merged**